runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -319,21 +319,26 @@ public void committed(long epoch, Layout layout, boolean force)
      *
      * @param endpoint Endpoint to bootstrap.
      * @param layout   Layout to bootstrap with.
-     * @return Completable Future which completes with True when the layout server is bootstrapped.
+     * @return True when the layout server is bootstrapped, false otherwise.
+     * @throws TimeoutException If a timeout has occurred.
      */
-    CompletableFuture<Boolean> bootstrapLayoutServer(@Nonnull String endpoint, @Nonnull Layout layout) {
-        return getRuntimeLayout(layout).getLayoutClient(endpoint).bootstrapLayout(layout)
-                .exceptionally(throwable -> {
-                    try {
-                        CFUtils.unwrap(throwable, AlreadyBootstrappedException.class);
-                    } catch (AlreadyBootstrappedException e) {
-                        log.info("bootstrapLayoutServer: Layout Server {} already bootstrapped.", endpoint);
-                    }
-                    return true;
-                })
-                .thenApply(result -> {
-                    log.info("bootstrapLayoutServer: Layout Server {} bootstrap successful", endpoint);
-                    return true;
-                });
+    boolean bootstrapLayoutServer(@Nonnull String endpoint, @Nonnull Layout layout) throws TimeoutException {
+        try {
+            CFUtils.getUninterruptibly(getRuntimeLayout(layout)
+                    .getLayoutClient(endpoint)
+                    .bootstrapLayout(layout),
+                    AlreadyBootstrappedException.class,
+                    TimeoutException.class);
+            log.info("bootstrapLayoutServer: Layout Server {} bootstrap successful.",
+                    endpoint);
+            return true;
+        } catch (AlreadyBootstrappedException abe) {
+            log.warn("bootstrapLayoutServer: Layout Server {} already bootstrapped.",
+                    endpoint);
+            return false;
+        } catch (TimeoutException te) {
+            log.error("bootstrapLayoutServer: Timeout occurred.");
+            throw te;
```

Comment:
Why specifically handling TimeoutException, there are other exceptions like NetworkException right? As explained in previous comment, the existing bootstrapLayoutServer implementation does return an exceptionally completed future due to CFUtil.unwrap(), I think the problem might not be here.

Reply:
Ok, I see. I won't touch it then. I'll leave management and layout views as is but will unwrap the exception in the AddNode sendRequest I guess.